biojava-structure/src/main/java/org/biojava/nbio/structure/SubstructureIdentifier.java

```diff
@@ -75,24 +82,27 @@
 	private static final Logger logger = LoggerFactory.getLogger(SubstructureIdentifier.class);
 
 
-	private final String pdbId;
+	private final PDBId pdbId;
 	private final List<ResidueRange> ranges;
 
 	/**
 	 * Create a new identifier from a string.
 	 * @param id
+	 * @throws PDBIdException 
 	 */
-	public SubstructureIdentifier(String id) {
+	public SubstructureIdentifier(String id) throws PDBIdException {
 		String[] idRange = id.split("\\.");
 		if(1 > idRange.length || idRange.length > 2 ) {
 			throw new IllegalArgumentException(String.format("Malformed %s: %s",getClass().getSimpleName(),id));
 		}
-		if(idRange[0].length() != 4) {
-			this.pdbId = idRange[0];
+		if(idRange[0].length() == 4) {  //Short form
+			this.pdbId = new PDBId(idRange[0]/* .toUpperCase() */);
+		} else if(idRange[0].length() == 12 && idRange[0].substring(0, 4).equalsIgnoreCase("PDB_")) { //extended form
+			this.pdbId = new PDBId(idRange[0]/* .toUpperCase() */);
+		} else {
+			this.pdbId = new PDBId(idRange[0]/* .toUpperCase() */);
 			// Changed from Exception to a warning to support files and stuff -sbliven 2015/01/22
 			logger.warn(String.format("Unrecognized PDB code %s",this.pdbId));
```

Comment:
I'm trying to remember why I changed it from an exception to a warning. I wonder if the StructureName tests break if this throws an exception for invalid PDBIds. Is this change why you had to change many of the tests to expect exceptions?

Either way, this if statement is unneeded. If we declare it to throw a PDBIdException then you can just do

```
this.pdbId = new PDBId(idRange[0]); // no if block, no warn statement.
```

Or  to preserve the old behavior, remove the `throws PDBIdExeption` from the declaration and do
```
try {
    this.pdbId = new PDBId(idRange[0]);
} catch(PDBIdException e) {
    logger.warn(...)
}

Reply:
My personal preference would be to support structures without PDBId where possible (probably why I went with the warning back in 2015)

Reply:
Done as possible. Please tell me your feedback when I submit the new commit.